RES-GUENTHER/README.md

```diff
@@ -140,6 +160,109 @@ Verwendung (Operation)
 P - Vor dem kritischen Abschnitt LOCK
 V - Nach dem kritischen Abschnitt UNLOCK
 
+### PIA 15/11/21
+
+<!-- md2apkg ignore-card -->
+
+Bekannte Fragen:
+
+**Rückgabewerte Fork**
+
+- Was gibt fork zurück, wenn Kind nicht gestartet hat?
+  - -1
+- fork() gibt 23 zurück - was ist passiert?
+  - Kind hat PID 23, wir sind Vater.
+- fork() gibt 0 zurück.
+  - Wir sind Kindprozess eines Fork.
+
+#### Zweck pipe
+
+IPC Prozess A -> B
+
+#### Wie kommt es, dass Sie von P. A zu Pr. B einen Kanal haben? Inbesondere mit fork()?
+
+<!-- TODO Frage klar definieren -->
+<!-- Antwort Studi: vgl. Sockets (File) -->
+
+- Kernel buffert das Geschriebene solange, bis es gelesen wird.
+
+#### Was passiert, wenn ich forke?
+
+- Ab fork() entsteht Kindprozess
+- Filedeskriptoren (auch Pipe), Variablen, etc. werden kopiert
+- Nicht kopiert werden: PID<!-- DUH -->
+
+$\rightarrow$ Prozesse, die verwandt sind, können über vor fork() angelegte Pipes kommunizieren.
+
+#### Welche Operationen auf einer Pipe kennen Sie? Beschreiben Sie deren Funktion
+
+- `write(pipefd[0],buffer)`
+  - Write: Aus Programm auf die Pipe schreiben, in den Kernelbuffer
+- `read(pipefd[1],buffer)`
+  - Read: Aus Kernelbuffer in Programmbuffer schreiben; aus Sicht des Programms von Pipe lesen
+
+#### Wie können Sie bidirektionale IPC ermöglichen?
+
+- Pipe umdrehen <!-- LOL -->, ggf. mit Zugriffskonzept (Semaphore)
+- 2 Pipes verwenden
+- Socket benutzen
+
+#### Was bedeutet close generell / bei Pipe?
+
+- Der Filedeskriptor wird gelöscht und steht nicht mehr zur Verfügung
+- Dadurch ist das dahinterliegende File (everything is a file) nicht mehr trivial erreichbar
+  - Es sei denn ein Prozess hat das File jetzt gerade noch geöffnet.
+
+Bei Pipes:
+
+- Seite einer Pipe schließen
+  - Schreibseite schließen: EOF senden
+
+<!-- **PIDs differieren stark, was ist passiert?**  -->
+
+<!-- #### Was passiert, wenn der Wertebereich für PIDs erschöpft ist? (2 Möglichkeiten) TODO -->
```

Comment:
Warum sind das Kommentare?

Reply:
Weil die Fragen sich mit den darauffolgenden vermischen, nochmal anschauen

Reply:
Ja gut, aber die Antwort auf die Erste fehlt - die wäre "Es wurden in der Zwischenzeit andere Prozesse **gestartet**"